From 6f1337393f3fb4e5f85bffc416da3e7d0ecbcf59 Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Thu, 29 Nov 2012 17:22:17 +0200
Subject: [PORT FROM MAIN] Telephony:Disable excessive MO SMS check-"eng" build

BZ: 61977

[DO NOT UPSTREAM]

Currently, there is a per-application based
SMS control, which checks the excessive
outgoing SMS usage. Upon limit reached,
an Alert dialog will be presented to the user
for confirmation of sending the SMS. Due to this
dialog, ACS test is blocked.

In engineering builds, this check will limit
the automated testing.

This patch disables the excessive outgoing
SMS check on "eng" build type only

Change-Id: I903ba63beeecebc52e0b275a82b73944dea5da9b
Orig-Change-Id: I3b4194be689e595c97e8072b4e09773aff818a94
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
Reviewed-on: http://android.intel.com:8080/77519
Reviewed-by: Fremaux, Pascal <pascal.fremaux@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Lucas, Guillaume <guillaume.lucas@intel.com>
Reviewed-by: Lebsir, SamiX <samix.lebsir@intel.com>
Tested-by: Lebsir, SamiX <samix.lebsir@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 .../android/internal/telephony/SMSDispatcher.java  |   11 +++++++----
 1 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/src/java/com/android/internal/telephony/SMSDispatcher.java b/src/java/com/android/internal/telephony/SMSDispatcher.java
index b4c0419..2babdfb 100644
--- a/src/java/com/android/internal/telephony/SMSDispatcher.java
+++ b/src/java/com/android/internal/telephony/SMSDispatcher.java
@@ -1001,10 +1001,13 @@ public abstract class SMSDispatcher extends Handler {
         // sending app is approved to send to short codes. Otherwise, a message is sent to our
         // handler with the SmsTracker to request user confirmation before sending.
         if (checkDestination(tracker)) {
-            // check for excessive outgoing SMS usage by this app
-            if (!mUsageMonitor.check(appInfo.packageName, SINGLE_PART_SMS)) {
-                sendMessage(obtainMessage(EVENT_SEND_LIMIT_REACHED_CONFIRMATION, tracker));
-                return;
+            String buildType = SystemProperties.get("ro.build.type", null);
+            if (buildType == null || !buildType.equals("eng")) {
+                // check for excessive outgoing SMS usage by this app
+                if (!mUsageMonitor.check(appInfo.packageName, SINGLE_PART_SMS)) {
+                    sendMessage(obtainMessage(EVENT_SEND_LIMIT_REACHED_CONFIRMATION, tracker));
+                    return;
+                }
             }
 
             int ss = mPhone.getServiceState().getState();
-- 
1.7.4.1

